R/civis_ml_utils.R

```diff
@@ -13,31 +13,65 @@ CIVIS_ML_CLASSIFIERS <- c("sparse_logistic", "gradient_boosting_classifier",
                          "extra_trees_classifier")
 
 #' @export
-print.civis_ml_classifier <- function(x, ...) {
-  class_names <- c(x$model_info$data$class_names)
-  tab <- with(x$metrics$metrics, t(cbind(roc_auc, p_correct)))
-  colnames(tab) <- class_names
-  rownames(tab) <- c("AUC", "Prop Correct")
-  print.civis_ml(x, tab = tab, ...)
+print.civis_ml_classifier <- function(x, digits = 4, ...) {
+  class_names <- get_model_data(x, "class_names")
+  aucs <- get_metric(x, "roc_auc")
+  p_cor <- get_metric(x, "p_correct")
+  dv_names <- get_model_data(x, "target_columns")
+
+  wfl <- model_workflow(x)
+  url <- model_url(x)
+  run_id <- x$run$id
+  job_id <- x$job$id
+
+  cat("<CivisML ", wfl, ">\n", sep = "")
+  cat(url, fill = TRUE)
+  cat("Job id: ", job_id, " Run id: ", run_id, "\n", fill = TRUE)
+
+  if (any(is_multiclass(x)) & is_multitarget(x)) {
+    tabs <- mapply(class_metric_table, p_cor, class_names, aucs)
+    } else if (is_multiclass(x)) {
+      tabs <- list(class_metric_table(p_cor, class_names, aucs))
+      } else {
+        cat("AUC: ", signif(aucs, digits = digits), fill = TRUE)
+        tabs <- list(class_metric_table(p_cor, class_names))
+      }
```

Comment:
small thing, but thought I mention because I was confused for a second:  Can we put the `else if` and `else` on the same indent level as `if` ?

Reply:
thanks for catching this